cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/OperationalStatsHttpHandler.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.handlers;
+
+import co.cask.cdap.common.NotFoundException;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.operations.OperationalStatsUtils;
+import co.cask.http.AbstractHttpHandler;
+import co.cask.http.HttpResponder;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+
+import java.util.Map;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * {@link co.cask.http.HttpHandler} for service provider statistics.
+ */
+@Path(Constants.Gateway.API_VERSION_3 + "/system/serviceproviders")
```

Comment:
Not all handlers are Singleton. What's our policy on it? This one in particular doesn't have any state, so it should be ok not to mark it?